mne/epochs.py

```diff
@@ -3669,38 +3693,70 @@ def _is_good(
+
     for refl, f, t in zip([reject, flat], [np.greater, np.less], ["", "flat"]):
         if refl is not None:
-            for key, thresh in refl.items():
+            for key, refl in refl.items():
+                criterion = refl
                 idx = channel_type_idx[key]
                 name = key.upper()
                 if len(idx) > 0:
                     e_idx = e[idx]
-                    deltas = np.max(e_idx, axis=1) - np.min(e_idx, axis=1)
                     checkable_idx = checkable[idx]
-                    idx_deltas = np.where(
-                        np.logical_and(f(deltas, thresh), checkable_idx)
-                    )[0]
+                    # Check if criterion is a function and apply it
+                    if callable(criterion):
+                        result = criterion(e_idx)
+                        _validate_type(result, tuple, result, "tuple")
+                        if len(result) != 2:
+                            raise TypeError(
+                                "Function criterion must return a " "tuple of length 2"
+                            )
+                        cri_truth, reasons = result
+                        _validate_type(cri_truth, (bool, np.bool_), cri_truth, "bool")
+                        _validate_type(
+                            reasons, (str, list, tuple), reasons, "str, list, or tuple"
+                        )
+                        idx_deltas = np.where(np.logical_and(cri_truth, checkable_idx))[
+                            0
+                        ]
+                    else:
+                        deltas = np.max(e_idx, axis=1) - np.min(e_idx, axis=1)
+                        idx_deltas = np.where(
+                            np.logical_and(f(deltas, criterion), checkable_idx)
+                        )[0]
 
                     if len(idx_deltas) > 0:
-                        bad_names = [ch_names[idx[i]] for i in idx_deltas]
-                        if not has_printed:
-                            logger.info(
-                                "    Rejecting %s epoch based on %s : "
-                                "%s" % (t, name, bad_names)
-                            )
-                            has_printed = True
-                        if not full_report:
-                            return False
+                        # Check to verify that refl is a callable that returns
+                        # (bool, reason). Reason must be a str/list/tuple.
+                        # If using tuple
+                        if callable(refl):
+                            if isinstance(reasons, (tuple, list)):
+                                for idx, reason in enumerate(reasons):
+                                    _validate_type(reason, str, reason, "str")
+                                bad_tuple += tuple(reasons)
+                            if isinstance(reasons, str):
+                                bad_tuple += (reasons,)
```

Comment:
A bit shorter. Adds a redundant check but fewer lines is generally nice even if it costs a tiny bit of recomputation. Also note the change in args to `_validate_type`
```suggestion
                            if isinstance(reasons, str):
                                reasons = (reasons,)
                            for idx, reason in enumerate(reasons):
                                _validate_type(reason, str, f"reasons[{idx}]")
                            bad_tuple += tuple(reasons)
```